6/6.2/6.2/main.cpp

```diff
@@ -0,0 +1,165 @@
+#include <iostream>
+
+using namespace std;
+
+int isNegative(int *number)
+{
+    if ((*number) > 0){
+        return 0;
+    }
+    else{
+        (*number) = (*number) * -1;
+        return 1;
+    }
+}
+
+void binPresentation(int *binNumber, int number, int maxLength)
+{
+    int count = 0;
+    int maxNumber = 128;
+    
+    for (int i = 1; i < maxNumber; i = i * 2)
+    {
+        if (number & i){
+            binNumber[maxLength - 1 - count] = 1;
+        }
+        else{
+            binNumber[maxLength - 1 - count] = 0;
+        }
+        count++;
+    }
+}
+
+void printBinNumber(int binNumber[], int maxLength)
+{
+    for (int i = 0; i < maxLength; i++)
+    {
+        cout << binNumber[i];
+    }
+}
+
+void sum(int *firstNumb, int *secondNumb, int *answer, int maxLength)
+{
+    int memory = 0;
+    for (int i = maxLength - 1; i >= 0; i--)
+    {
+        if (memory == 0)
+        {
+            if (firstNumb[i] + secondNumb[i] == 2)
+            {
+                memory = 1;
+                answer[i] = 0;
+            }
+            else{
+                answer[i] = firstNumb[i] + secondNumb[i];
+            }
+        }
+        else{
+            if (firstNumb[i] + secondNumb[i] == 2)
+            {
+                memory = 1;
+                answer[i] = 1;
+            }
+            else{
+                if (firstNumb[i] + secondNumb[i] == 1){
+                    answer[i] = 0;
+                    memory = 1;
+                }
+                else{
+                    answer[i] = 1;
+                    memory = 0;
+                }
+            }
+        }
+    }
+}
+
+void makeAdditionalCode(int *binNumber, int *binPresentationOfOne, int maxLength)
+{
+    for (int i = 1; i < maxLength; i++)
+    {
+        if (binNumber[i] == 1){
+            binNumber[i] = 0;
+        }
+        else{
+            binNumber[i] = 1;
+        }
+    }
+    sum(binNumber, binPresentationOfOne, binNumber, maxLength);
+}
+
+void fromBinToDecimal(int binNumber[], int maxLength)
+{
+    int answer = 0;
+    int powerOfTwo = 1;
+    
+    for (int i = maxLength - 1; i >= 1; i--)
+    {
+        answer = powerOfTwo * binNumber[i] + answer;
+        powerOfTwo = powerOfTwo * 2;
+    }
+
+    cout << answer;
+}
+
+int main()
```

Comment:
The program sums two numbers and outputs them in binary form
Enter first number : 11
Enter second number : -13
Binary representation of first number : 00001011
Binary representation of second number : 10001101
Binary representation answer : 62997691111110
Sum : 126